### ISSUE TITLE
Proposal: use something like `ld`'s `--just-symbols` option from zig build

### ISSUE THREAD

**Issue by Khitiara** [ziglang/zig#22112]
The `--just-symbols` option for `ld` (alias `-R`) specifies a file to be included containing data with the following format (spaces required):
```
symbol_name1 = address1;
symbol_name2 = address2;
```
allowing extra symbols to be included in linking which are then resolvable from `extern` and equivalent as with other linker-defined/linkerscript symbols. 
the main uses of something like this (or the equivalent single-symbol `--defsym symbol=address`) is to refer symbolically to absolute addresses defined in other binaries, e.g. in a multi-stage OS loader in osdev or other similar multi-stage binary loading scheme that cannot use dynamic linking.

**Comment by alexrp** [comment]
Would something along the lines of this work?

https://github.com/ziglang/zig/blob/aa7d138462602e086aacf738e4b92bfa3372bebe/lib/std/os/windows/tls.zig#L15-L18

**Comment by Khitiara** [comment]
ive never seen that syntax before and cannot find any documentation for it in asm, but if it works by setting in that example `__tls_array` to be at address `0x2C` then it might work yes. I'd still like an option to use the linker option if possible for parity with existing tooling, but that will at least work as a workaround for my uses I think

**Comment by Leandros** [comment]
@alexrp
> Would something along the lines of this work?

Yes, this is a feasible work around to defining those in a build script.
